src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/judge/JudgeServiceTest.java

```diff
@@ -0,0 +1,80 @@
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.referencedata.Judge;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.referencedata.JudgeEmploymentStatus;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.repository.JudgeRepository;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.never;
+
+
+class JudgeServiceTest {
+    private JudgeRepository judgeRepository;
+    private final String judgeCode = "testCode";
+    private final String judgeName = "testName";
+    private final String tribunalOffice = "Aberdeen";
+
+    @BeforeEach
+    void setup() {
+        judgeRepository = mock(JudgeRepository.class);
+    }
+
+    @Test
+    void shouldSaveJudge() {
+        var adminData = createAdminData(judgeCode, judgeName, tribunalOffice, "SALARIED");
+        var addJudgeService = new JudgeService(judgeRepository);
+        when(judgeRepository.existsByCodeAndTribunalOffice(judgeCode, TribunalOffice.valueOfOfficeName(tribunalOffice))).thenReturn(false);
+        when(judgeRepository.existsByNameAndTribunalOffice(judgeName,
+                TribunalOffice.valueOfOfficeName(tribunalOffice))).thenReturn(false);
+
+        assertDoesNotThrow(() -> addJudgeService.saveJudge(adminData));
+        verify(judgeRepository, times(1)).save(createJudge(adminData));
+    }
+
+    @Test
+    void shouldReturnErrorIfJudgeWithSameCodeAndOfficeExists() {
+        var adminData = createAdminData(judgeCode, judgeName, tribunalOffice, "SALARIED");
+        var addJudgeService = new JudgeService(judgeRepository);
+        when(judgeRepository.existsByCodeAndTribunalOffice(judgeCode,TribunalOffice.valueOfOfficeName(tribunalOffice))).thenReturn(true);
+        assertThrows(SaveJudgeException.class, () -> addJudgeService.saveJudge(adminData));
+        verify(judgeRepository, never()).save(createJudge(adminData));
+    }
+
+    @Test
+    void shouldReturnErrorIfJudgeWithSameNameAndOfficeExists() {
+        var adminData = createAdminData(judgeCode, judgeName, tribunalOffice, "SALARIED");
+        var addJudgeService = new JudgeService(judgeRepository);
+        when(judgeRepository.existsByCodeAndTribunalOffice(judgeCode,TribunalOffice.valueOfOfficeName(tribunalOffice))).thenReturn(false);
+        when(judgeRepository.existsByNameAndTribunalOffice(judgeName,
+                TribunalOffice.valueOfOfficeName(tribunalOffice))).thenReturn(true);
+        assertThrows(SaveJudgeException.class, () -> addJudgeService.saveJudge(adminData));
+        verify(judgeRepository, never()).save(createJudge(adminData));
+    }
+
+    private AdminData createAdminData(String judgeCode, String judgeName, String tribunalOffice, String employmentStatus) {

```

Comment:
Please refactor to introduce a Judge class passed around between methods, instead of many primitive type params.